frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsRow.tsx

```diff
@@ -3,70 +3,75 @@ import { TableRow, TableData, RowFunction } from '@console/internal/components/f
 import { ResourceLink, Timestamp, Kebab, ResourceKebab } from '@console/internal/components/utils';
 import { referenceForModel } from '@console/internal/module/k8s';
 import { TaskRunKind } from '../../../utils/pipeline-augment';
-import { TaskRunModel, PipelineModel, TaskModel } from '../../../models';
+import { TaskRunModel, PipelineModel, TaskModel, ClusterTaskModel } from '../../../models';
 import { tableColumnClasses } from './taskruns-table';
 import { Status } from '@console/shared';
 import { pipelineRunFilterReducer as taskRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
 
 const taskRunsReference = referenceForModel(TaskRunModel);
-const taskReference = referenceForModel(TaskModel);
 const pipelineReference = referenceForModel(PipelineModel);
 
-const TaskRunsRow: RowFunction<TaskRunKind> = ({ obj, index, key, style, ...props }) => (
-  <TableRow id={obj.metadata.uid} index={index} trKey={key} style={style}>
-    <TableData className={tableColumnClasses[0]}>
-      <ResourceLink
-        kind={taskRunsReference}
-        name={obj.metadata.name}
-        namespace={obj.metadata.namespace}
-        title={obj.metadata.name}
-        data-test-id={obj.metadata.name}
-      />
-    </TableData>
-    <TableData className={tableColumnClasses[1]} columnID="namespace">
-      <ResourceLink kind="Namespace" name={obj.metadata.namespace} />
-    </TableData>
-    {props.customData?.showPipelineColumn && (
-      <TableData className={tableColumnClasses[2]}>
-        {obj.metadata.labels['tekton.dev/pipeline'] ? (
+const TaskRunsRow: RowFunction<TaskRunKind> = ({ obj, index, key, style, ...props }) => {
+  const taskReference =
+    obj.spec.taskRef?.kind === ClusterTaskModel.kind
+      ? referenceForModel(ClusterTaskModel)
+      : referenceForModel(TaskModel);
```

Comment:
Can we keep the result of referenceForModel outside of this Row function? So that it's not required to recalculate the kind for each row? It keeps the references consts also at one place.

Reply:
done. PTAL